python/torch_mlir_e2e_test/test_suite/rng.py

```diff
@@ -76,9 +76,64 @@ def forward(self, x, y, z):
 @register_test_case(module_factory=lambda: UniformStaticModule())
 def UniformStaticModule_basic(module, tu: TestUtils):
     module.forward(
-        tu.rand(256, 512, 8).double(),
-        tu.rand(512, 1024, 4).double(),
-        tu.rand(512, 256, 4).double())
+        tu.rand(256, 512, 12).double(),
+        tu.rand(512, 1024, 12).double(),
+        tu.rand(512, 256, 12).double())
+
+# ==============================================================================
+
+class UniformNoCorrelationModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    def correlation(self, stack):
+        """Calculate the correlation of the two rows in `stack`.
+
+        TODO: Remove this by adding support for `torch.corrcoef`.
+        """
+        m = stack - torch.mean(stack, dim=1, keepdim=True)
+        cov = torch.matmul(m, m.t()) / (stack.size()[1] - 1)
+        return cov[0, 1] / torch.sqrt(cov[0, 0] * cov[1, 1])
+
+    @export
+    @annotate_args([
+        None,
+        ([2000000], torch.float64, True),
+        ([2000000], torch.float64, True),
+    ])
+    def forward(self, x, y):
+        # Correlation of two independent uniforms
+        a = torch.ops.aten.uniform(x)
+        b = torch.ops.aten.uniform(y)
+        stack = torch.cat((a.unsqueeze(0), b.unsqueeze(0)))
+        corr_a_b = self.correlation(stack)
+
+        # Split first dimension of large buffer
+        larger = torch.empty((2,) + x.size(), dtype=torch.float64)
+        larger = torch.ops.aten.uniform(larger)
+        corr_major = self.correlation(larger)
+
+        # Split second dimension of large buffer
+        corr_minor = self.correlation(larger.t().reshape(2, -1))
+
+        # This is hacky. The problem with returning just the correlations
+        # is that `torch.allclose` becomes stricter the closer values are to
+        # zero. Since the correlations are on the order of 1E-3, `rtol=1E-3`,
+        # and `atol=1E-7`, the correlations have to have a difference smaller
+        # than `atol + rtol * correlation = 1E-6`, which is too strict.
+        # By subtracting from 1, the correlations are expected to have a
+        # difference of around 1E-3.
```

Comment:
another possibly equally hacky (but perhaps more clear, and decouple it from the global rtol)is something like:

```
torch.where(torch.abs(corr) < 0.001, 1, 2)
```

Reply:
That's so much better, thanks!